src/main/java/StringManipulation.java

```diff
@@ -1,33 +1,74 @@
 public class StringManipulation implements StringManipulationInterface {
+    private String string;
 
     @Override
     public String getString() {
-        return null;
+
+        return string;
     }
 
     @Override
     public void setString(String string) {
+        this.string = string;
     }
 
     @Override
     public int count() {
-        return 0;
+        if (string==null) {
+
+            return 0;
+        }
+        String[] words = string.split("\\s+");
+        return words.length;
     }
 
     @Override
     public String removeNthCharacter(int n, boolean maintainSpacing) {
-        return null;
+        if (string == null || string.isEmpty()) {
+            return "";
+        }
+        StringBuilder result = new StringBuilder();
+        for (int i = 0; i < string.length(); i++) {
+            char c = string.charAt(i);
+            if ((i + 1) % n != 0) {
+                if (maintainSpacing || !Character.isWhitespace(c)) {
+                    result.append(c);
+                }
+            }
+        }
+        return result.toString();
     }
 
     @Override
     public String[] getSubStrings(int startWord, int endWord) {
-        return null;
+        if (string == null || string.isEmpty()) {
+            return new String[0];
+        }
+        String[] words = string.trim().split("\\s+");
+        if (startWord < 0 || startWord >= words.length || endWord < 0 || endWord >= words.length || startWord > endWord) {
+            return new String[0];
```

Comment:
You could also throw an indexOutofBounds exception here to help with troubleshooting. But returning an empty array is also fine.